client.go

```diff
@@ -0,0 +1,88 @@
+package rest
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	// ErrServiceUnavailable represents a 422
```

Comment:
If these map directly to HTTP errors and you're not adding any context, I'd just return the original HTTP error

Reply:
That sounds good, I can just return the error body instead of logging it like I do in debugFailedResponse